frontend/javascripts/oxalis/view/right-border-tabs/segments_tab/segments_view.tsx

```diff
@@ -209,6 +218,45 @@ const mapStateToProps = (state: OxalisState): StateProps => {
   };
 };
 
+const getOrInitializeSelectedSegmentsOrGroup = (
+  visibleSegmentationLayer: APISegmentationLayer | null | undefined,
+  state: OxalisState,
+  visibleSegments: SegmentMap | null | undefined,
+  visibleSegmentGroups: Array<SegmentGroup>,
+) => {
+  const nothingSelectedObject = { segments: [], group: null };
+  if (visibleSegmentationLayer == null) {
+    return nothingSelectedObject;
+  }
+  const { segments, group } =
+    state.localSegmentationData[visibleSegmentationLayer.name].selectedIds;
+  if (segments.length === 0 && group == null) return nothingSelectedObject;
+
+  // Ensure that the ids of previously selected segments are removed
+  // if these segments don't exist anymore.
+  const currentSegmentIds = new Set(visibleSegments?.map((segment) => segment.id));
+  const currentGroupIdsWithoutRootGroup = new Set(
+    visibleSegmentGroups.map((group) => group.groupId),
+  );
+  const currentGroupIds = currentGroupIdsWithoutRootGroup.add(MISSING_GROUP_ID);
+  const selectedIds = {
+    segments: segments.filter((id) => currentSegmentIds.has(id)),
+    group: group != null && currentGroupIds.has(group) ? group : null,
+  };
+  const haveSegmentsOrGroupsBeenRemovedFromList =
+    selectedIds.segments.length !== segments.length || selectedIds.group !== group;
+  if (haveSegmentsOrGroupsBeenRemovedFromList) {
+    Store.dispatch(
```

Comment:
cool, that you found a way to keep this logic incoporporated :+1: I think, it would be even cleaner if the potential removal of selected ids would happen in the reducer. However, the current way probably works okay, too.